src/test/java/org/elasticsearch/test/store/MockFSDirectoryService.java

```diff
@@ -89,7 +89,7 @@ public void beforeIndexShardClosed(ShardId sid, @Nullable IndexShard indexShard,
                             // so that even in tests where don't flush we can check the integrity of the Lucene index
                             if (indexShard.engine().hasUncommittedChanges()) { // only if we have any changes
                                 logger.info("{} flushing in order to run checkindex", indexShard.shardId());
-                                Releasables.close(indexShard.engine().snapshotIndex()); // Keep translog for tests that rely on replaying it
+                                Releasables.close(indexShard.engine().snapshotIndex(true)); // Keep translog for tests that rely on replaying it
```

Comment:
Passing true here means that a flush will happen first, but the comment says that the translog will be kept, which one is intended?